code/scripting/api/libs/ui.cpp

```diff
@@ -542,6 +558,208 @@ ADE_FUNC(getCmdBriefing,
+	mission_goal_fail_incomplete();
+	hud_config_as_player();
+	debrief_init();
+
+	API_Access = false;
+	
+	return ade_set_args(L, "b", true);
+}
+
+ADE_FUNC(getDebriefingMusicName,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Gets the file name of the music file to play for the debriefing.",
+	"string",
+	"The file name or empty if no music")
+{
+	return ade_set_args(L, "s", common_music_get_filename(debrief_select_music()).c_str());
+}
+
+ADE_FUNC(getDebriefing, l_UserInterface_Debrief, nullptr, "Get the debriefing", "debriefing", "The debriefing data")
+{
+	// get a pointer to the appropriate debriefing structure
+	if (MULTI_TEAM) {
+		return ade_set_args(L, "o", l_Debrief.Set(Net_player->p_info.team));
+	} else {
+		return ade_set_args(L, "o", l_Debrief.Set(0));
+	}
+}
+
+ADE_FUNC(getEarnedMedal,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the earned medal name and bitmap",
+	"string, string",
+	"The name and bitmap or NIL if not earned")
+{
+	char filename[80];
+	SCP_string displayname;
+
+	if (Player->stats.m_medal_earned != -1) {
+		debrief_choose_medal_variant(filename,
+			Player->stats.m_medal_earned,
+			Player->stats.medal_counts[Player->stats.m_medal_earned] - 1);
+		displayname = Medals[Player->stats.m_medal_earned].get_display_name();
+
+		return ade_set_args(L, "ss", displayname, filename);
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(getEarnedPromotion,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the earned promotion stage, name, and bitmap",
+	"debriefing_stage, string, string",
+	"The promotion stage, name and bitmap or NIL if not earned")
+{
+	char filename[80];
+	SCP_string displayname;
+
+	if (Player->stats.m_promotion_earned != -1) {
+		Promoted = Player->stats.m_promotion_earned;
+		debrief_choose_medal_variant(filename, Rank_medal_index, Promoted);
+		displayname = Ranks[Promoted].name;
+
+		return ade_set_args(L, "oss", l_DebriefStage.Set(debrief_stage_h(&Promotion_stage)), displayname, filename);
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(getEarnedBadge,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the earned badge stage, name, and bitmap",
+	"debriefing_stage, string, string",
+	"The badge stage, name and bitmap or NIL if not earned")
+{
+	char filename[80];
+	SCP_string displayname;
+
+	if (Player->stats.m_badge_earned.size()) {
+		debrief_choose_medal_variant(filename,
+			Player->stats.m_badge_earned.back(),
+			Player->stats.medal_counts[Player->stats.m_badge_earned.back()] - 1);
+		displayname = Ranks[Promoted].name;
+
+		return ade_set_args(L, "oss", l_DebriefStage.Set(debrief_stage_h(&Badge_stage)), displayname, filename);
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(getTraitor,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the traitor stage",
+	"debriefing_stage",
+	"The traitor stage or nil if the player is not traitor. Will also clear out any earned stats if true")
+{
+	if (Turned_traitor) {
+		Player->flags &= ~PLAYER_FLAGS_PROMOTED;
+		scoring_level_init(&Player->stats);
```

Comment:
I would move the clearing of stats to another function.  By convention, `get` and `is` functions are not supposed to have side effects.